onnx/test/parser_test.py

```diff
@@ -268,6 +268,52 @@ def test_parse_various_float_values(self, test_literal, expect_exception):
                 str(model.graph.node[0].attribute[0].f), str(float(test_literal))
             )
 
+    @parameterized.expand(
+        [
+            ("bfloat16", TensorProto.BFLOAT16),
+            ("bool", TensorProto.BOOL),
+            ("complex64", TensorProto.COMPLEX64),
+            ("complex128", TensorProto.COMPLEX128),
+            ("double", TensorProto.DOUBLE),
+            ("float16", TensorProto.FLOAT16),
+            ("float", TensorProto.FLOAT),
+            ("float8e4m3fn", TensorProto.FLOAT8E4M3FN),
+            ("float8e4m3fnuz", TensorProto.FLOAT8E4M3FNUZ),
+            ("float8e5m2", TensorProto.FLOAT8E5M2),
+            ("float8e5m2fnuz", TensorProto.FLOAT8E5M2FNUZ),
+            ("int4", TensorProto.INT4),
+            ("int8", TensorProto.INT8),
+            ("int16", TensorProto.INT16),
+            ("int32", TensorProto.INT32),
+            ("int64", TensorProto.INT64),
+            ("string", TensorProto.STRING),
+            ("uint4", TensorProto.UINT4),
+            ("uint8", TensorProto.UINT8),
+            ("uint16", TensorProto.UINT16),
+            ("uint32", TensorProto.UINT32),
+            ("uint64", TensorProto.UINT64),
+        ]
+    )
+    def test_parse_graph_types(self, name, itype) -> None:
+        cast = """
+           <
+             ir_version: 10,
+             opset_import: [ "" : 19]
+           >
+           agraph (float[N] X) => (__STYPE__[N] C)
+           <
+             __STYPE__[1] weight = {0}
+           >
+           {
+              C = Cast<to=__ITYPE__>(X)
+           }
+           """
+        text_graph = cast.replace("__ITYPE__", str(itype)).replace("__STYPE__", name)
```

Comment:
This can be done with str.format() and named arguments

Reply:
done